File: DAM1/T6/EjerciciosClase/src/Ejercicio1/model/Persona.java
package Ejercicio1.model;

public class Persona {

    private String nombre, dni;

    public Persona(){}
    public Persona(String nombre, String dni) {
        this.nombre = nombre;
        this.dni = dni;
    }

    public void mostrarDatos(){
        System.out.println("Esta persona se llama: " + nombre);
        System.out.println("Su dni es: " + dni);
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }
}

File: DAM2/PMDM/T1/TrabajadoresHerencia/src/model/Persona.kt
package model

abstract class Persona (var nombre : String, var apellido : String, var dni : String){

    var telefono : Int? = null
    var correo : String? = null

    constructor(nombre: String, apellido: String, dni: String, telefono : Int, correo : String) : this(nombre, apellido, dni) {
        this.telefono = telefono
        this.correo = correo
    }

    open fun mostrarDatos(){
        println("nombre = ${nombre}")
        println("apellido = ${apellido}")
        println("dni = ${dni}")
        println("telefono = ${telefono?: "no se ha definido"}")
        println("correo = ${correo ?: "no se ha definido"}")
    }
}